presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java

```diff
@@ -183,16 +182,11 @@ private void openRowGroup()
     {
         // read the dictionary
         if (!dictionaryOpen && dictionarySize > 0) {
-            if (dictionary.length < dictionarySize) {
-                dictionary = new long[dictionarySize];
+            GetDictionaryResult getDictionaryResult = dictionaryProvider.getDictionary(streamDescriptor, dictionary, dictionarySize);
+            dictionary = getDictionaryResult.getDictionaryBuffer();
+            if (getDictionaryResult.getIsBufferOwner()) {
```

Comment:
This logic has been add to `getRetainedSizeInBytes`. Can we address that?

Reply:
As far as I understand, `getRetainedSizeInBytes` for long dictionary batch stream readers or long batch stream readers are not used anywhere, and particularly not used for memory accounting. I could add an additional field in batch stream readers like we did to selective stream reader, but this is all temporary change before we centralize memory accounting in the dictionary provider. Thoughts?

Reply:
That is not correct. `getRetainedSizeInBytes` is the keep function we use to do memory accounting.

Reply:
+1 to what James said, getRetainedSizeInBytes is the key function used for memory accounting.

Reply:
I understand that is the case for selective reader. Intellij was telling me it's not used anywhere in the project. Maybe it wasn't able to look at super class usage? I'm changing it just in case. Updating soon.